portal/server/controllers/dashboard/reports.controller.js

```diff
@@ -0,0 +1,73 @@
+const downloadCsv = require('../../utils/downloadCsv');
+const api = require('../../api');
+
+exports.getPortalReports = async (req, res) => {
+  const { userToken } = req.session;
+
+  const facilities = await api.getUnissuedFacilitiesReports(userToken);
+
+  const pastDeadlineUnissuedFacilitiesCount = facilities.filter((facility) => facility.daysLeftToIssue < 0);
+  const facilitiesThatNeedIssuingCount = facilities.filter((facility) => facility.daysLeftToIssue < 15 && facility.daysLeftToIssue >= 0);
+  if (facilities) {
+    return res.render('reports/reports-dashboard.njk', {
+      allUnissuedFacilitiesCount: facilities.length,
+      pastDeadlineUnissuedFacilitiesCount: pastDeadlineUnissuedFacilitiesCount.length,
+      facilitiesThatNeedIssuingCount: facilitiesThatNeedIssuingCount.length,
+      user: req.session.user,
+      primaryNav: 'reports',
+    });
+  }
+  return res.redirect('/not-found');
+};
+
+exports.getUnissuedFacilitiesReports = async (req, res) => {
+  const { userToken } = req.session;
+
+  const facilities = await api.getUnissuedFacilitiesReports(userToken);
+
+  if (facilities) {
+    return res.render('reports/unissued-facilities.njk', { facilities, user: req.session.user, primaryNav: 'reports' });
+  }
+
+  return res.redirect('/not-found');
+};
+
+exports.downloadUnissuedFacilitiesReports = async (req, res) => {
+  const { userToken } = req.session;
+  const facilities = await api.getUnissuedFacilitiesReports(userToken);
+
+  // not all columns are needed, in which case, the array below will specify
+  // the properties that we want to add to the CSV file and the label for them
```

Comment:
what does the first part mean please, `not all columns are needed`, not sure why/the meaning - looks like all the columns are needed? 🙃